File: analytics/metrics-platform/src/test/java/org/wikimedia/metrics_platform/config/StreamConfigTest.java
package org.wikimedia.metrics_platform.config;

import static org.assertj.core.api.Assertions.assertThat;
import static org.wikimedia.metrics_platform.config.sampling.SampleConfig.Identifier.SESSION;

import java.util.Set;

import org.junit.jupiter.api.Test;
import org.wikimedia.metrics_platform.json.GsonHelper;

import com.google.gson.Gson;

class StreamConfigTest {

    @Test void testStreamConfigDeserialization() {
        Gson gson = GsonHelper.getGson();
        String streamConfigJson = "{\"stream\":\"test.event\",\"schema_title\":\"test/event\"," +
                "\"destination_event_service\":\"eventgate-logging-local\"," +
                "\"producers\":" +
                "{\"metrics_platform_client\":{\"provide_values\":[\"page_id\",\"user_id\"]}}," +
                "\"sample\":{\"rate\":0.5,\"identifier\":\"session\"}}";
        StreamConfig streamConfig = gson.fromJson(streamConfigJson, StreamConfig.class);
        assertThat(streamConfig.getStreamName()).isEqualTo("test.event");
        assertThat(streamConfig.getSchemaTitle()).isEqualTo("test/event");
        assertThat(streamConfig.getSampleConfig().getRate()).isEqualTo(0.5);
        assertThat(streamConfig.getSampleConfig().getIdentifier()).isEqualTo(SESSION);
        assertThat(streamConfig.getProducerConfig().getMetricsPlatformClientConfig().getRequestedValues())
                .isEqualTo(Set.of("page_id", "user_id"));
        assertThat(streamConfig.getDestinationEventService()).isEqualTo(DestinationEventService.LOCAL);
    }
}

File: analytics/testkitchen/src/test/java/org/wikimedia/testkitchen/config/StreamConfigTest.kt
package org.wikimedia.testkitchen.config

import org.junit.Assert.assertEquals
import org.junit.Test
import org.wikimedia.testkitchen.JsonUtil
import org.wikimedia.testkitchen.config.sampling.SampleConfig

internal class StreamConfigTest {
    @Test
    fun testStreamConfigDeserialization() {
        val streamConfigJson = "{\"stream\":\"test.event\",\"schema_title\":\"test/event\"," +
                "\"destination_event_service\":\"eventgate-logging-local\"," +
                "\"producers\":" +
                "{\"metrics_platform_client\":{\"provide_values\":[\"page_id\",\"user_id\"]}}," +
                "\"sample\":{\"rate\":0.5,\"identifier\":\"session\"}}"
        val streamConfig: StreamConfig = JsonUtil.decodeFromString(streamConfigJson)!!

        assertEquals("test.event", streamConfig.streamName)
        assertEquals("test/event", streamConfig.schemaTitle)
        assertEquals(0.5, streamConfig.sampleConfig!!.rate, 0.0000001)
        assertEquals(SampleConfig.UNIT_SESSION, streamConfig.sampleConfig!!.unit)
        assertEquals(listOf("page_id", "user_id"), streamConfig.producerConfig!!.metricsPlatformClientConfig!!.requestedValues)
        assertEquals(DestinationEventService.LOCAL, streamConfig.destinationEventService)
    }
}
